client/src/components/ContentNavigator/convert.ts

```diff
@@ -0,0 +1,201 @@
+      UI_PROP_DF_EXECUTION_ORDERED: "false",
+    },
+    links: [],
+    nodes: {},
+    parameters: {},
+    connections: [],
+    extendedProperties: {},
+    stickyNotes: [],
+  };
+
+  for (let idx = 0; idx < inputList.length; idx++) {
+    const idNode = `id-${Date.now()}-${Math.floor(Math.random() * 10000)}`;
+    arrayIdNode.push(idNode);
+    const idNote = `id-${Date.now()}-${Math.floor(Math.random() * 10000)}`;
+    const entry = inputList[idx];
+    const propPorts = getPropPort(idx, inputList);
+    const stepNode = {
+      nodeType: "step",
+      version: 1,
+      id: idNode,
+      name: stepTitle[entry.language],
+      note: {
+        version: 1,
+        id: idNote,
+        name: null,
+        description: null,
+        properties: {
+          UI_NOTE_PROP_HEIGHT: "0",
+          UI_NOTE_PROP_IS_EXPANDED: "false",
+          UI_NOTE_PROP_IS_STICKYNOTE: "false",
+          UI_NOTE_PROP_WIDTH: "0",
+        },
+      },
+      priority: idx,
+      properties: {
+        UI_PROP_COLORGRP: "0",
+        UI_PROP_IS_INPUT_EXPANDED: "false",
+        UI_PROP_IS_OUTPUT_EXPANDED: "false",
+        UI_PROP_NODE_DATA_ID: stepRef[entry.language],
+        UI_PROP_NODE_DATA_MODIFIED_DATE: nowTimestamp,
+        UI_PROP_XPOS: ((idx + 1) * NODE_SPACING).toString(),
+        UI_PROP_YPOS: "75",
+      },
+      portMappings: [
+        {
+          mappingType: "tableStructure",
+          portIndex: 0,
+          portName: "outTables",
+          tableStructure: {
+            columnDefinitions: null,
+          },
+        },
+      ],
+      stepReference: {
+        type: "uri",
+        path: `/dataFlows/steps/${stepRef[entry.language]}`,
+      },
+      arguments: {
+        codeOptions: {
+          code: entry.code,
+          contentType: "embedded",
+          logHTML: "",
+          resultsHTML: "",
+          variables: [
+            {
+              name: "_input1",
+              value: {
+                portIndex: 0,
+                portName: "inTables",
+                referenceType: "inputPort",
+              },
+            },
+            {
+              name: "_output1",
+              value: {
+                arguments: {},
+                portIndex: 0,
+                portName: "outTables",
+                referenceType: "outputPort",
+              },
+            },
+          ],
+        },
+      },
+    };
+    stepNode.properties = { ...stepNode.properties, ...propPorts };
+    flowData.nodes[idNode] = stepNode;
+
+    if (idx > 0) {
+      const connection = {
+        sourcePort: {
+          node: arrayIdNode[idx - 1],
+          portName: "outTables",
+          index: 0,
+        },
+        targetPort: {
+          node: arrayIdNode[idx],
+          portName: "inTables",
+          index: 0,
+        },
+      };
+      flowData.connections.push(connection);
+    }
+  }
+  return flowData;
+}
+
+export function convertSASNotebookToFlow(
```

Comment:
I guess it would be great if a test can be created for this, so that there's a clear view of the expected input/output.

Reply:
Yeah, I'd love to see more testing in this PR

Reply:
I have added some tests in eec49d4ec0f52dc16f9836aebfd4b92839892cfa.
Please let me know if this is the kind of tests you have in mind ?

Tests:
- Converting SAS Notebooks to code lists: multi cell + single cell
- Converting code lists to Flows: swimlane mode + node mode / multi cell + single cell

Some of the challenges encountered:
- handling special characters before parsing
- removing unique ids, timestamps and dates from comparison